packages/transaction-manager/src/transaction-manager.ts

```diff
@@ -236,31 +227,144 @@ export default class TransactionManager implements TransactionTypes.ITransaction
+        }
+      } catch (error) {
+        console.warn(`Failed to decrypt channel ${channelId}:`, error);
+        result.transactions[channelId] = [];
+        result.ignoredTransactions[channelId] = [];
+      }
+    }
 
-        accumulator.transactions[channelId] = cleaned.transactions;
-        accumulator.ignoredTransactions[channelId] = cleaned.ignoredTransactions;
-        return accumulator;
-      },
-      Promise.resolve({ transactions: {}, ignoredTransactions: {} }),
+    // Apply pagination only if both page and pageSize are defined
+    let channelsToProcess = validChannels;
+    let paginationMeta;
+
+    if (page !== undefined && pageSize !== undefined) {
+      const startIndex = (page - 1) * pageSize;
+      const endIndex = Math.min(startIndex + pageSize, validChannels.length);
+      channelsToProcess = validChannels.slice(startIndex, endIndex);
+
+      paginationMeta = {
+        total: validChannels.length,
+        page,
+        pageSize,
+        hasMore: endIndex < validChannels.length,
+      };
+    }
+
+    // Process the selected channels
+    await Promise.all(
+      channelsToProcess.map(async (channelId) => {
+        const cleaned = await this.channelParser.decryptAndCleanChannel(
+          channelId,
+          resultGetTx.result.transactions[channelId],
+        );
+
+        result.transactions[channelId] =
+          cleaned.transactions as unknown as TransactionTypes.ITransaction[];
+        result.ignoredTransactions[channelId] =
+          cleaned.ignoredTransactions as unknown as TransactionTypes.ITransaction[];
+      }),
     );
 
+    // Create the final meta object
+    const successfulChannelIds = Object.keys(result.transactions);
+    const paginatedMeta = {
+      ...resultGetTx.meta,
+      storageMeta: Object.keys(resultGetTx.meta.storageMeta || {})
+        .filter((key) => successfulChannelIds.includes(key))
+        .reduce((acc, key) => {
+          acc[key] = resultGetTx.meta.storageMeta?.[key] || [];
+          return acc;
+        }, {} as Record<string, StorageTypes.IEntryMetadata[]>),
+      transactionsStorageLocation: Object.keys(resultGetTx.meta.transactionsStorageLocation)
+        .filter((key) => successfulChannelIds.includes(key))
+        .reduce((acc, key) => {
+          acc[key] = resultGetTx.meta.transactionsStorageLocation[key];
+          return acc;
+        }, {} as Record<string, string[]>),
+      pagination: paginationMeta,
+    };
+
     return {
       meta: {
-        dataAccessMeta: resultGetTx.meta,
-        ignoredTransactions: result.ignoredTransactions,
+        dataAccessMeta: paginatedMeta,
+        ignoredTransactions: result.ignoredTransactions as unknown as Record<
+          string,
+          TransactionTypes.IIgnoredTransaction[]
+        >,
+      },
+      result: {
+        transactions: result.transactions as unknown as Record<
+          string,
+          TransactionTypes.ITimestampedTransaction[]
+        >,
```

Comment:
Why do you need to cast `as unknown` here?

Reply:
Good catch. Fixed it!